core/src/main/java/org/eclipse/hono/util/KubernetesContainerUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
```

Comment:
IMHO we should indicate that this class has been copied from the Apache Log4J 2 project so that it becomes more obvious, why we mention that project in the NOTICE file. I would also opt for moving this class to the service-base module, as it seems to be relevant for Hono services and adapters only, isn't it?

Reply:
> IMHO we should indicate that this class has been copied from the Apache Log4J 2 project so that it becomes more obvious, why we mention that project in the NOTICE file.

You mean putting such a note directly below the copyright header? I've put it in the class javadoc, like we did for other copied classes.

Reply:
My bad, didn't see that. That is exactly what I meant ...

Reply:
> I would also opt for moving this class to the service-base module, as it seems to be relevant for Hono services and adapters only, isn't it?

Since the `KubernetesContainerUtil` class is currently used from a class in the `hono-client-command` module, that would mean adding a dependency to `hono-service-base` there, which doesn't look right.

Reply:
I see